src/driver.rs

```diff
@@ -385,23 +376,18 @@ pub fn main() {
                 || arg_value(&orig_args, "--cap-lints", |val| val == "allow").is_none();
 
             if clippy_enabled {
-                args.extend_from_slice(&["--cfg".to_owned(), r#"feature="cargo-clippy""#.to_owned()]);
+                args.extend(vec!["--cfg".into(), r#"feature="cargo-clippy""#.into()]);
                 if let Ok(extra_args) = env::var("CLIPPY_ARGS") {
-                    args.extend(extra_args.split("__CLIPPY_HACKERY__").filter_map(|s| {
-                        if s.is_empty() {
-                            None
-                        } else {
-                            Some(s.to_string())
-                        }
-                    }));
+                    match extra_args.strip_suffix("__CLIPPY_HACKERY__") {
```

Comment:
This isn't the same, is it? This assumes that `__CLIPPY_HACKERY__` is at the end of the string, while the prior version just assumes that it is anywhere in the string.

Reply:
removed.